### PR TITLE
Fixing a text editing bug happening when text field changes.

### PR DESCRIPTION
**Pull request by nturgut**
During text editing, if one changes between text fields, sometimes the size/transform was not being updated. 

## Description

This was due to the check in this editable text:

if (size != _lastSize || transform != _lastTransform) 

When there is a new connection lastSize and lastTransform was not being cleared.

By carrying lastSize,lastTransform to TextInputConnection we are making sure that when there is a new connection, the old values won't be cached.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -678,14 +681,18 @@ class TextInputConnection {
...
-      'TextInput.setEditableSizeAndTransform',
-      <String, dynamic>{
-        'width': editableBoxSize.width,
-        'height': editableBoxSize.height,
-        'transform': transform.storage,
-      },
-    );
+    if (editableBoxSize != _cachedSize || transform != _cachedTransform) {
+      _cachedSize = editableBoxSize;
+      _cachedTransform = transform;
```
Comment (yjbanov) [inline packages/flutter/lib/src/services/text_input.dart]: Wondering if we should store a copy of `transform`. Unfortunately, `Matrix4` is mutable.

**Review by yjbanov — APPROVED** [review]

File: packages/flutter/lib/src/services/text_input.dart (hunk 1 above)
Comment (nturgut) [reply packages/flutter/lib/src/services/text_input.dart]: done.  thanks for the suggestion

**Comment by nturgut** [comment]
The IOS platform errors seems irrelevant. I'm merging the PR.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -2215,6 +2215,88 @@ void main() {
...
+      isMethodCall('TextInput.setEditableSizeAndTransform', arguments: <String, dynamic>{
+        'width': 800,
+        'height': 14,
+        'transform': Matrix4.identity().storage.toList(),
+      }),
+    );
+
+    // Move to the next editable text.
+    await tester.showKeyboard(find.byKey(ValueKey<String>(controller2.text)));
+    final MethodCall methodCall2 = log.firstWhere((MethodCall m) => m.method == 'TextInput.setEditableSizeAndTransform');
```
Comment (mdebbar) [inline packages/flutter/test/widgets/editable_text_test.dart]: @nturgut  sorry I didn't review this earlier.

`log.firstWhere(...)` always returns the first platform message that was sent when the first input field was focused. In order to make the test really work, we should clear `log` before switching to another input field.

**Pull request merged**